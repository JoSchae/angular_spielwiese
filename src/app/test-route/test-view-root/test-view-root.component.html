<!-- <tmp-login></tmp-login> -->

<router-outlet></router-outlet>
