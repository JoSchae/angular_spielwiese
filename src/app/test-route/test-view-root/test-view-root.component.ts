import { Component, OnInit } from '@angular/core';

@Component({
  selector: 'tmp-test-route-root',
  templateUrl: './test-view-root.component.html',
  styleUrls: ['./test-view-root.component.scss']
})
export class TestViewRootComponent implements OnInit {

  constructor() { }

  ngOnInit() {
  }

}
