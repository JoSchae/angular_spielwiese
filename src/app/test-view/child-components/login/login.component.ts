import { Component, OnInit } from '@angular/core';

import { Observable } from 'rxjs';
import { IAuthenticationState } from '../../../_stores/authentication/auth.store.service';
import { AuthenticationFacade } from '../../../_facades/authentication/authentication.facade';

@Component({
    selector: 'tmp-login',
    template: '<tmp-login-view [isLoggedIn]="(authState | async).isLoggedIn" (buttonEmitter)="onOutputRecived($event)"></tmp-login-view>',
})
export class LoginComponent {

    public authState: Observable<IAuthenticationState>;

    constructor(private _authenticationFacade: AuthenticationFacade) {
        this.authState = this._authenticationFacade.authenticationState$;
    }

    public onOutputRecived(event: string) {
        if (event === 'login') {
            this._authenticationFacade.login();
        } else {
            this._authenticationFacade.logout();
        }
    }

}
