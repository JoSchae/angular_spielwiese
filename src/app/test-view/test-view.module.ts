import { NgModule } from '@angular/core';
import { HttpClientModule } from '@angular/common/http';
import { TestViewRoutingModule } from './test-view-routing.module';
import { TestViewRootComponent } from './test-view-root/test-view-root.component';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { LoginComponent } from './child-components/login/login.component';
import { LoginViewComponent } from './child-components/login/login-view/login-view.component';
import { BrowserModule } from '@angular/platform-browser';
import { FacadesModule } from '../_facades/facades.module';
import { AuthenticationFacade } from '../_facades/authentication/authentication.facade';

@NgModule({
    declarations: [
        TestViewRootComponent,
        LoginComponent,
        LoginViewComponent
    ],
    imports: [
        BrowserModule,
        CommonModule,
        FormsModule,
        FacadesModule,
        TestViewRoutingModule,
        HttpClientModule
    ],
    providers: [
        AuthenticationFacade
    ]
})
export class TestViewModule { }
