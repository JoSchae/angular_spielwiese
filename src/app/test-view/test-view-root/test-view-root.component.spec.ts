import { TestViewRootComponent } from './test-view-root.component';
import { SpectatorRouting, createRoutingFactory, Spectator, createComponentFactory } from '@ngneat/spectator/jest';
import { LoginComponent } from '../child-components/login/login.component';
import { LoginViewComponent } from '../child-components/login/login-view/login-view.component';
import { TestViewModule } from '../test-view.module';
import { TestViewRoutingModule } from '../test-view-routing.module';
import { RouterTestingModule } from '@angular/router/testing';

describe('TestViewRootComponent', () => {

    let spectator: SpectatorRouting<TestViewRootComponent>;
    const createComponent = createRoutingFactory({
        component: TestViewRootComponent,
        imports: [
            TestViewModule,
            RouterTestingModule
        ],
        declareComponent: false,
        // stubsEnabled: false,
        // routes: [
        //     {
        //         path: 'test',
        //         children: [
        //             { path: '', redirectTo: 'login', pathMatch: 'full' },
        //             { path: 'login', component: LoginComponent }
        //         ]
        //     }
        // ]
    });

    beforeEach(() => {
        spectator = createComponent();
    });

    it('should create', () => {
        expect(spectator.component).toBeTruthy();
    });

    it('should have correct initial snapshot', () => {
        expect(spectator.fixture).toMatchSnapshot();
    });
});
