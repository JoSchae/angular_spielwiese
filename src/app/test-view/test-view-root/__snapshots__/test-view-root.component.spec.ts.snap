// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestViewRootComponent should have correct initial snapshot 1`] = `
<tmp-test-view-root>
  <router-outlet />
</tmp-test-view-root>
`;
